src/core/Akka.Streams/Dsl/Source.cs

```diff
@@ -1136,6 +1136,62 @@ public static class Source
         {
             return FromGraph(new UnfoldResourceSourceAsync<T, TSource>(create, read, close));
         }
+        
+        /// <summary>
+        /// Unfolds a resource, using ValueTask returns instead of normal tasks,
+        /// As well as optimizing when the result successfully completed synchronously.
+        /// <para/>
+        /// There is an overload that takes a create State,
+        /// allowing one to minimize captures and leakage
+        /// 
+        ///   (essentially, pass initial state in,
+        ///    then any additional state gets passed into initial TState) 
+        /// </summary>
+        /// <param name="create">function that is called on stream start and creates/opens resource.</param>
+        /// <param name="read">function that reads data from opened resource. It is called each time backpressure signal
+        /// is received. Stream calls close and completes when <see cref="Task"/> from read function returns None.</param>
+        /// <param name="close">function that closes resource</param>
+        /// <typeparam name="T"></typeparam>
+        /// <typeparam name="TSource"></typeparam>
+        /// <returns>A source of T that is created on stream start and read on backpressure</returns>
+        public static Source<T, NotUsed> UnfoldResourceValueTaskAsync<T,
```

Comment:
Nitpick / question: can we call this `UnfoldResourceAsync` and overload the other call or is there going to be some conflict with `Task` / `ValueTask` conversion?

Reply:
Mayyyyybe?

Biggest concern off the top of my head, is that it may be easy to accidentally wind up with the `Task` variant, i.e.

```
Source.UnfoldResourceAsync(
async () => inThing.OpenAsync(), //Is Task
async (r) => r.ReadAsync(), //Is ValueTask
async r => r.DisposeAsync() // is ValueTask
```
Above should(?) give a compiler error.
But below, need to verify it would not wind up using `Task` variant:
```
Source.UnfoldResourceAsync(
async () => return await inThing.OpenAsync(), //Is Task
async (r) => return await r.ReadAsync(), //Is ValueTask
async r => return await r.DisposeAsync() // is ValueTask
```